### PR TITLE
Add basic parser for Outpost24 scan format

### PR DESCRIPTION
**Pull request by jvz**
@madchap @jeffret-b @Wadeck @daniel-beck

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes must include the necessary migrations in the dojo/dd_migrations folder.
- [x] Add applicable tests to the unit tests.

### CODE REVIEW

**Comment by Maffooch** [comment]
Don't forget about adding the fixture :)

**Comment by jvz** [comment]
Alright, I've addressed those issues. There's still a little bit of code to add for the endpoints part, though I'll follow up with that later today. Let's make sure this part passes CI at least.

**Comment by jvz** [comment]
Seems to be a test failure still. I'll update later today.

**Comment by jvz** [comment]
Figured it out. @jeffret-b feel free to review now!

File: dojo/unittests/test_outpost24_parser.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from django.test import TestCase
+from dojo.tools.outpost24.parser import Outpost24Parser
+from dojo.models import Test
+
+
+class TestOutpost24Parser(TestCase):
```
Comment (jeffret-b) [inline dojo/unittests/test_outpost24_parser.py]: Does it make sense to have at least one test that check the finding details?

**Comment by jeffret-b** [comment]
> Remaining to be done before merge

@jvz is this comment in other original description still accurate? Is there still remaining work to be done?

File: dojo/unittests/test_outpost24_parser.py (hunk 1 above)
Comment (jvz) [reply dojo/unittests/test_outpost24_parser.py]: You'd think so, but I didn't see any other tests doing that. 🤷‍♂

Comment (jeffret-b) [reply dojo/unittests/test_outpost24_parser.py]: I agree that it doesn't seem to be common in these tests, so maybe it isn't valuable. I did find this example, though, that is doing something along those lines: https://github.com/DefectDojo/django-DefectDojo/blob/ac0502498b467247ad237c5668a868f681d188ca/dojo/unittests/test_checkmarx_parser.py#L165

**Comment by jvz** [comment]
When you merge this, please squash the history. Thanks!

**Comment by madchap** [comment]
@jvz if you could upload that sample file as well in the sibling repo at https://github.com/DefectDojo/sample-scan-files, it would be great. Thanks!

**Comment by ptrovatelli** [comment]
looks pretty neat. 
was this sample.xml file manually modified? there are a lot of duplicates for example `<id>1377090</id>` appearing twice with just `firstseen` changed
if this is a `regular` report, it might be good to do some aggregation upon import; it'll greatly improve the performances by inserted much less lines into the db and also will make the analysis easier.

dependency check amongst other scanners do this, using a dict. ideally update nb_occurences in that case. 

it might be nice to use that falsepositive flag too to fill in the according field in the db

**Comment by jvz** [comment]
> was this sample.xml file manually modified? there are a lot of duplicates for example `<id>1377090</id>` appearing twice with just `firstseen` changed

Yes, I started from a real scan, removed several things, and didn't really update any summary statistics at the top as I wasn't sure how they're calculated by the tool itself. I'm not sure exactly what the `id` element corresponds to, but it's a tool-specific id.

**Pull request merged**